### ISSUE TITLE
I can not compile for my Linux 4K satellite box 😭 (armv7l)

### ISSUE THREAD

**Issue by AnharMiah** [wasmerio/wasmer#1481]
### Motivation

Since one of the **core** purposes of `WASM` and by extension (`WASI`) is to carry the noble "torch" from the JVM's dream of _"thou shall write once and thou shall runneth everywhere"_  🤔

😍 Excited by the possibility I went ahead and downloaded the `wamser` source and then using `cross` I attempted to to compile for my satellite box:

```shell
$ cd wasmer
$ cross build --target cross build --target armv7-unknown-linux-gnueabihf
```  
And away it went and I drifted off day dreaming about the possibilities; but then... 😱

```shell
   Compiling cfg-if v0.1.10

<SNIPPED FOR BREVITY>

   Compiling generate-wasi-tests v0.17.0 (/project/tests/generate-wasi-tests)
error[E0425]: cannot find function `get_fault_info` in this scope
   --> lib/runtime-core/src/fault.rs:289:21
    |
289 |         let fault = get_fault_info(siginfo as _, ucontext);
    |                     ^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `ARCH` in this scope
   --> lib/runtime-core/src/fault.rs:295:87
    |
295 |                         if let Some(x) = v.runnable_module.get_inline_breakpoint_size(ARCH) {
    |                                                                                       ^^^^ not found in this scope
    |
help: possible candidate is found in another module, you can import it into scope
    |
30  | use std::env::consts::ARCH;
    |

error[E0425]: cannot find value `ARCH` in this scope
   --> lib/runtime-core/src/fault.rs:304:29
    |
304 | ...                   ARCH,
    |                       ^^^^ not found in this scope
    |
help: possible candidate is found in another module, you can import it into scope
    |
30  | use std::env::consts::ARCH;
    |

error: unused import: `Architecture`
   --> lib/runtime-core/src/fault.rs:273:26
    |
273 |     use crate::backend::{Architecture, InlineBreakpointType};
    |                          ^^^^^^^^^^^^
    |
note: the lint level is defined here
   --> lib/runtime-core/src/lib.rs:19:5
    |
19  |     unused_imports,
    |     ^^^^^^^^^^^^^^

   Compiling wasmer-bin v0.17.0 (/project)
   Compiling wabt v0.9.2
error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0425`.
error: could not compile `wasmer-runtime-core`.

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

## **TL:DR**

Please can we have `wasmer` for more targets e.g like `armv7l` _(or is that even possible?)_

**Comment by Hywan** [comment]
Still relevant.

**Comment by Hywan** [comment]
I've tried to cross-compile from macOS to arm-linux-gnueabihf. I confirm that it's not a supported platform by `wasmer-vm`. Updating https://github.com/wasmerio/wasmer/blob/c7f4bd28109ee1eaa18c5979c5abe6ada5beb67c/lib/vm/src/trap/traphandlers.rs#L205 should make it work I guess, maybe there is more, not sure yet.

**Comment by AnharMiah** [comment]
thanks @Hywan might give it another go!
